### ISSUE TITLE
Clicking "Messages" when not logged in causes Error

### ISSUE THREAD

**Issue by Rvice** [Admidio/admidio#1703]
**Describe the bug**
Accessing "Messages" when not logged in throws an exception

**To Reproduce**
1. Go to home page in Incognito Mode
2. Click on 'Messages
3. See error

**Expected behavior**
I'd expect to see the same behavior as in the Admidio playground. Although, in our case, we would not want anonymous sending of messages to our instance.

**Screenshots**
![image](https://github.com/user-attachments/assets/d85c5615-7d7f-4966-8445-bb01d457aa72)

**System (please complete the following information):**
 - Admidio-Version: 4.3.12
 - PHP: 7.4.10
 - Database: MySQL 5.6.51-91.0-log
 - Browser: Chrome
 - Browser-Version: Version 131.0.6778.70 (Official Build) (64-bit)

**Additional context**
I have not checked the messages while not logged in and just discovered it. It's possible this issue predated 4.3.12 and just never observed it. I did go through and delete the `adm_program` folder as outlined in the upgrade instructions (auto-update is apparently not available on my host)

**Comment by Fasse** [comment]
I could not reproduce your problem. Could you please describe step by step how to get that error at https://www.admidio.org/demo_en

**Comment by Rvice** [comment]
Interesting, in playground i see similar to my site:
![image](https://github.com/user-attachments/assets/1b1c79f1-1417-4e84-91b5-328b13a2ac2c)

However, on the Demo, I don't see the "Messages" link in the navbar
![image](https://github.com/user-attachments/assets/5f771feb-5407-4825-bf70-c4275a173f39)

Also, locked out for 15 minutes due to a bad password to try and get the Messages link setup on the Demo site

**Comment by Fasse** [comment]
Is the behavior reproducable at the playground?

**Comment by Rvice** [comment]
I do not see the same behavior. The following is the debug log when accessing the page in anonymous mode. The error is not present when logged in:

`Admidio.ERROR: REDIRECT: Header already sent! `

```
[2024-11-22 12:14:24.435961] Admidio.INFO: ##################################################################################################  {"file":"./admidio/adm_program/system/bootstrap/logging.php","line":56,"class":null,"callType":null,"function":"initLogging"}
[2024-11-22 12:14:24.438143] Admidio.INFO: URL: https://www.admidio.org/demo_en_Test/adm_program/modules/messages/messages.php  {"file":"./admidio/adm_program/system/bootstrap/logging.php","line":57,"class":null,"callType":null,"function":"initLogging"}
[2024-11-22 12:14:24.438182] Admidio.INFO: CONSTANTS: URLS & PATHS & FOLDERS {"VERSIONS":{"ADMIDIO":"4.3.12","PHP":"7.4.10"},"SCHEME":"https","HTTPS":true,"PORT":443,"HOST":"www.admidio.org","DOMAIN":"www.admidio.org","ADMIDIO_URL_PATH":"","URLS":{"ADMIDIO_URL":"https://www.admidio.org/demo_en_Test","FILE_URL":"https://www.admidio.org/demo_en_Test/adm_program/modules/messages/messages.php","CURRENT_URL":"https://www.admidio.org/demo_en_Test/adm_program/modules/messages/messages.php"},"PATHS":{"SERVER_PATH":"./admidio","ADMIDIO_PATH":"./admidio","CURRENT_PATH":"./admidio/adm_program/modules/messages/messages.php"}} {"file":"./admidio/adm_program/system/bootstrap/logging.php","line":82,"class":null,"callType":null,"function":"initLogging"}
[2024-11-22 12:14:24.443777] Admidio.DEBUG: DATABASE: Create DB-Instance with default params! {"engine":"mysql","host":"robertviceipage45214.ipagemysql.com","port":null,"name":"db_name","username":"db_username","password":"******"} {"file":"./admidio/adm_program/system/classes/Database.php","line":137,"class":"Database","callType":"::","function":"createDatabaseInstance"}
[2024-11-22 12:14:24.443842] Admidio.DEBUG: DATABASE: DSN-String: "mysql:host=mysql.admido.org;dbname=db_name;charset=utf8"!  {"file":"./admidio/adm_program/system/classes/Database.php","line":890,"class":"Database","callType":"->","function":"setDSNString"}
[2024-11-22 12:14:24.446546] Admidio.INFO: SQL: SET SQL_MODE = 'ANSI'  {"file":"./admidio/adm_program/system/classes/Database.php","line":774,"class":"Database","callType":"->","function":"queryPrepared"}
[2024-11-22 12:14:24.446855] Admidio.DEBUG: SQL: Execution time 0.246048 ms  {"file":"./admidio/adm_program/system/classes/Database.php","line":789,"class":"Database","callType":"->","function":"queryPrepared"}
[2024-11-22 12:14:24.446887] Admidio.INFO: SQL: SET SQL_BIG_SELECTS = 1  {"file":"./admidio/adm_program/system/classes/Database.php","line":774,"class":"Database","callType":"->","function":"queryPrepared"}
[2024-11-22 12:14:24.447111] Admidio.DEBUG: SQL: Execution time 0.205040 ms  {"file":"./admidio/adm_program/system/classes/Database.php","line":789,"class":"Database","callType":"->","function":"queryPrepared"}
[2024-11-22 12:14:24.447132] Admidio.DEBUG: DATABASE: connected!  {"file":"./admidio/adm_program/system/classes/Database.php","line":171,"class":"Database","callType":"->","function":"__construct"}
[2024-11-22 12:14:24.447155] Admidio.INFO: SQL: SHOW COLUMNS FROM adm_sessions  {"file":"./admidio/adm_program/system/classes/Database.php","line":719,"class":"Database","callType":"->","function":"query"}
[2024-11-22 12:14:24.449693] Admidio.DEBUG: SQL: Execution time 2.507925 ms  {"file":"./admidio/adm_program/system/classes/Database.php","line":730,"class":"Database","callType":"->","function":"query"}
[2024-11-22 12:14:24.478448] Admidio.INFO: Session Started! {"name":"ADMIDIO_db_name_adm_SESSION_ID","limit":0,"path":"/","domain":"www.admidio.org","secure":true,"httpOnly":true,"sameSite":"lax","sessionId":"<session id is here>"} {"file":"./admidio/adm_program/system/classes/Session.php","line":569,"class":"Session","callType":"::","function":"start"}
[2024-11-22 12:14:24.478569] Admidio.INFO: SQL: SELECT * FROM adm_sessions WHERE ses_id = ? [2938] {"file":"./admidio/adm_program/system/classes/Database.php","line":774,"class":"Database","callType":"->","function":"queryPrepared"}
[2024-11-22 12:14:24.478998] Admidio.INFO: SQL: Found rows: 1  {"file":"./admidio/adm_program/system/classes/Database.php","line":785,"class":"Database","callType":"->","function":"queryPrepared"}
[2024-11-22 12:14:24.479056] Admidio.DEBUG: SQL: Execution time 0.434160 ms  {"file":"./admidio/adm_program/system/classes/Database.php","line":789,"class":"Database","callType":"->","function":"queryPrepared"}
[2024-11-22 12:14:24.483182] Admidio.INFO: SQL: UPDATE adm_sessions SET ses_timestamp = ?, ses_reload = ? WHERE ses_id = ? ["2024-11-22 12:14:24",0,2938] {"file":"./admidio/adm_program/system/classes/Database.php","line":774,"class":"Database","callType":"->","function":"queryPrepared"}
[2024-11-22 12:14:24.483973] Admidio.DEBUG: SQL: Execution time 0.702858 ms  {"file":"./admidio/adm_program/system/classes/Database.php","line":789,"class":"Database","callType":"->","function":"queryPrepared"}
[2024-11-22 12:14:24.485246] Admidio.ERROR: REDIRECT: Header already sent! {"url":"https://www.admidio.org/demo_en_Test/adm_program/modules/messages/messages_write.php?mode=1","statusCode":303} {"file":"./admidio/adm_program/system/bootstrap/function.php","line":606,"class":null,"callType":null,"function":"admRedirect"}
[2024-11-22 12:14:24.511146] Admidio.INFO: SHUTDOWN {"execution_time":"132.178068 ms","memory_usage":"5.452 MiB"} {"file":"./admidio/adm_program/system/bootstrap/shutdown.php","line":21,"class":null,"callType":null,"function":"admShutdown"}
[2024-11-22 12:14:24.512165] Admidio.DEBUG: DATABASE: sleep/serialize!  {"file":"./admidio/adm_program/system/classes/Database.php","line":184,"class":"Database","callType":"->","function":"__sleep"}
```
Note: I generalized the db/URL/domain info/file paths for privacy.

**Comment by Fasse** [comment]
I found the issue and could fix it for version 4.3.13 .

**Comment by Rvice** [comment]
Confirmed, put the change in the instance and it works now.
